packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/WorkflowService.java

```diff
@@ -663,6 +664,21 @@ public void updateNodeStatus(final Workflow workflow, final Map<UUID, String> st
 		}
 	}
 
+	public void addOrUpdateAnnotation(final Workflow workflow, final WorkflowAnnotation annotation) {
+		if (workflow.getAnnotations() == null || workflow.getAnnotations().isEmpty()) {
+			workflow.setAnnotations(new HashMap<UUID, WorkflowAnnotation>());
+		}
+		if (annotation.getId() == null) {
```

Comment:
Just to confirm that hibernate isn't doing this automagically, and this won't clash with that?

Reply:
Correct. WorkflowAnnotation is packaged into Workflow object. The unique-id is here is to make CRUD operations easier to manage, since deletion is possible using a pure-index approach is not reliable.